### DISCUSSION TITLE
Read_time modifier working with bard that has sets?

### DISCUSSION THREAD

**Discussion by austriker27** [statamic/cms#4341] [Troubleshooting]
Hello! 

Has anyone gotten the `read_time` modifier working with bard when you have sets? 

I found this post for awhile ago but its not working for me (post is from V2):
https://statamic.com/forum/2970-read-time-from-bard-fieldtype

When I try this in antlers: 

```twig
{{ blog_content | flatten | join | strip_tags | read_time:180 }} min read
```

I get this error: 
`Error: Array to string conversion`

Inside the CMS, at the bottom of the bard field it does show the read time when Im using sets. Not sure if this is helpful but I looked at the [source code](https://github.com/statamic/cms/blob/a3d9bbd5ac9bbd4ffad0d9126dd75f4756b6c570/resources/js/components/fieldtypes/bard/BardFieldtype.vue#L125) and its using this package: https://www.npmjs.com/package/read-time-estimate

**Comment by mynetx** [comment]
@austriker27 I ran into the same issue and ended up writing a modifier that would try to flatten everything. But it's far from working well, so I'd prefer a core fix.

I've [opened an issue](https://github.com/statamic/cms/issues/4343) for you.

```php
<?php

namespace App\Modifiers;

use Statamic\Modifiers\Modifier;

class TextOnly extends Modifier
{
    /**
     * Modify a value.
     *
     * @param mixed  $value    The value to be modified
     * @param array  $params   Any parameters used in the modifier
     * @param array  $context  Contextual values
     * @return mixed
     */
    public function index($value, $params, $context)
    {
        $value = array_flatten(json_decode(json_encode($value), false));

        $fields = ['lead', 'title', 'description', 'content'];
        $text = '';
        foreach ($value as $item) {
            foreach ($fields as $field) {
                if (isset($item->$field) && is_string($item->$field)) {
                    $text .= $item->$field . ' '; 
                }
            }
        }

        return strip_tags($text);
    }
}
```

I’m using the modifier this way in my Antlers template:

```antlers
{{ newsbuilder | text_only | strip_tags | read_time }}&nbsp;min read.
```

**Reply by austriker27** [reply]
Nice, this is useful until its available in core. Thanks for sharing!

**Reply by austriker27** [reply]
@mynetx I tried this approach but am getting "0" min read. Sorry I don't really know PHP so I'm not sure where my issue is. But I created the modifier `TextOnly` and added `blog_content` (because thats my bard field name) to the following line:

```php
$fields = ['lead', 'title', 'description', 'content', 'blog_content'];
```

Then used it like this in the HTML:
```twig
{{ blog_content | text_only | strip_tags | read_time:180 }}&nbsp;min read
````

Am I missing anything? Thanks!

**Reply by mynetx** [reply]
@austriker27 It should not be the name of the Bard field, but the name of the content field inside any of your sets￼. And the code does not support nested sets. ￼I’m thinking about a more generic solution as we speak￼.

**Reply by austriker27** [reply]
> @austriker27 It should not be the name of the Bard field, but the name of the content field inside any of your sets￼. And the code does not support nested sets. ￼I’m thinking about a more generic solution as we speak￼.

Gotcha gotcha! Thanks, I've got it working now! Wooohooo!

**Reply by mynetx** [reply]
@austriker27 Just came up with a slightly modified version that will walk into nested structures. Maybe that helps?

```php
<?php

namespace App\Modifiers;

use Statamic\Modifiers\Modifier;

class TextOnly extends Modifier
{
    /**
     * Modify a value.
     *
     * @param mixed  $value    The value to be modified
     * @param array  $params   Any parameters used in the modifier
     * @param array  $context  Contextual values
     * @return mixed
     */
    public function index($value, $params, $context)
    {
        $value = array_flatten(json_decode(json_encode($value), false));

        return $this->_walk($value);
    }

    protected function _walk($value)
    {
        $fields = ['lead', 'title', 'description', 'content', 'text', 'copy'];
        $text = '';
        foreach ($value as $item => $value) {
            if (is_string($value)) {
                if (in_array($item, $fields)) {
                    $text .= $value . ' ';
                }
                continue;
            }
            if (is_array($value) || is_object($value)) {
                $text .= $this->_walk((array) $value);
            }
        }
        return strip_tags($text);
    }
}
```

**Comment by duncanmcclean** [comment]
Not sure what the gents think but in an ideal world I'd love if you could just do this... (the `read_time` modifier would know about Bard and how to make it into something that's usuable)

```antlers
{{ blog_content | read_time:180 }} min read
```

**Reply by mynetx** [reply]
That’d be my ideal world, too… 👼🏻

**Comment by jasonvarga** [comment]
Keep an eye on https://github.com/statamic/ideas/issues/94

**Reply by austriker27** [reply]
Oh awesome, thanks Jason!
